src/backend/Orders/Jobs/RecalcOrdersPercentage/RecalcOrdersPercentageJob.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Core.Database;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Orders.Jobs.RecalcOrdersPercentage
+{
+    public class RecalcOrdersPercentageJob
+    {
+        private readonly AppDbContext _appDbContext;
+        private readonly RecalcOrdersPercentageJobDataService _recalcOrdersPercentageJobDataService;
+        private readonly RecalcOrdersPercentageJobService _recalcOrdersPercentageJobService;
+
+        public RecalcOrdersPercentageJob(
+            RecalcOrdersPercentageJobDataService recalcOrdersPercentageJobDataService,
+            RecalcOrdersPercentageJobService recalcOrdersPercentageJobService,
+            AppDbContext appDbContext)
+        {
+            _recalcOrdersPercentageJobDataService = recalcOrdersPercentageJobDataService;
+            _recalcOrdersPercentageJobService = recalcOrdersPercentageJobService;
+            _appDbContext = appDbContext;
+        }
+
+        public async Task<ActionResult> Execute()
+        {
+            var listOfOrders = await _recalcOrdersPercentageJobDataService.Fetch();
+
+            if (listOfOrders != null)
```

Comment:
To nigdy nie będzie prawdą. Wywal

Reply:
w sumie to powinno być listOfOrders.Any()... 
ale masz rację, mimo wszystko foreach nie wykona się gdy lista jest pusta.